framework/test/jest/unit/specs/modules/chain/dpos/apply.spec.js

```diff
@@ -0,0 +1,446 @@
+				expect.anything(),
+			);
+
+			expect(
+				stubs.storage.entities.Account.increaseFieldBy,
+			).not.toHaveBeenCalledWith(expect.any(Object), 'missedBlocks');
+			expect(
+				stubs.storage.entities.Account.increaseFieldBy,
+			).not.toHaveBeenCalledWith(expect.any(Object), 'voteWeightReceived');
+			expect(stubs.storage.entities.Account.update).not.toHaveBeenCalled();
+		});
+	});
+
+	describe('Given block is the last block of the round', () => {
+		let lastBlockOfTheRound;
+		let feePerDelegate;
+		let rewardPerDelegate;
+		let totalFee;
+		let getTotalEarningsOfDelegate;
+		beforeEach(() => {
+			// Arrange
+			lastBlockOfTheRound = {
+				height: 202,
+				generatorPublicKey: delegateWhoForgedLast.publicKey,
+			};
+
+			delegateAccounts.forEach(account => {
+				when(stubs.storage.entities.Account.get)
+					.calledWith({
+						publicKey_eq: account.publicKey,
+					})
+					.mockResolvedValue(account);
+			});
+
+			feePerDelegate = randomInt(10, 100);
+			totalFee = feePerDelegate * constants.ACTIVE_DELEGATES;
+
+			// Delegates who forged got their rewards
+			rewardPerDelegate = randomInt(1, 20);
+
+			getTotalEarningsOfDelegate = account => {
+				const blockCount = delegatesWhoForged.filter(
+					d => d.publicKey === account.publicKey,
+				).length;
+				const reward = rewardPerDelegate * blockCount;
+				const fee = feePerDelegate * blockCount;
+				return {
+					reward,
+					fee,
+				};
+			};
+
+			stubs.storage.entities.Round.summedRound.mockResolvedValue([
+				{
+					fees: totalFee, // dividable to ACTIVE_DELEGATE count
+					rewards: delegatesWhoForged.map(() => rewardPerDelegate),
+					delegates: delegatesWhoForged.map(account => account.publicKey),
+				},
+			]);
+		});
+
+		it('should increase "missedBlocks" field by "1" for the delegates who missed forging slot', async () => {
+			// Act
+			await dpos.apply(lastBlockOfTheRound, stubs.tx);
+
+			// Assert
+			expect(
+				stubs.storage.entities.Account.increaseFieldBy,
+			).toHaveBeenCalledWith(
+				{
+					publicKey_in: delegatesWhoMissed.map(a => a.publicKey),
+				},
+				'missedBlocks',
+				'1',
+				stubs.tx,
+			);
+		});
+
+		it('should distribute rewards and fees ONLY to the delegates who forged', async () => {
+			// Act
+			await dpos.apply(lastBlockOfTheRound, stubs.tx);
+
+			// Assert
+			expect.assertions(constants.ACTIVE_DELEGATES);
+
+			// Assert Group 1/2
+			uniqueDelegatesWhoForged.forEach(account => {
+				expect(stubs.storage.entities.Account.update).toHaveBeenCalledWith(
+					{
+						publicKey_eq: account.publicKey,
+					},
+					expect.any(Object),
+					stubs.tx,
+				);
+			});
+
+			// Assert Group 2/2
+			delegatesWhoMissed.forEach(account => {
+				expect(stubs.storage.entities.Account.update).not.toHaveBeenCalledWith({
+					publicKey_eq: account.publicKey,
+				});
+			});
+		});
+
+		it('should distribute more rewards and fees (with correct balance) to delegates based on number of blocks they forged', async () => {
```

Comment:
it's probably good to add the test where
`should distrubute reward and fee for delegate who forged once but missed once`